searx/engines/quark.py

```diff
@@ -0,0 +1,293 @@
+        matches = re.findall(pattern, text, re.DOTALL)
+
+        for match in matches:
+            data = json.loads(match)
+            initial_data = data.get('data', {}).get('initialData', {})
+            extra_data = data.get('extraData', {})
+
+            source_category = extra_data.get('sc')
+            source_category_parsers = {
+                'addition': parse_addition,
+                'ai_page': parse_ai_page,
+                'baike_sc': parse_baike_sc,
+                'finance_shuidi': parse_finance_shuidi,
+                'life_show_general_image': parse_life_show_general_image,
+                'nature_result': parse_nature_result,
+                'news_uchq': parse_news_uchq,
+                'ss_note': parse_ss_note,
+                'ss_pic': parse_ss_pic_text,
+                'ss_text': parse_ss_pic_text,
+                'travel_dest_overview': parse_travel_dest_overview,
+                'travel_ranking_list': parse_travel_ranking_list,
+            }
+
+            parsers = source_category_parsers.get(source_category)
+            if parsers:
+                parsed_results = parsers(initial_data)
+                if isinstance(parsed_results, list):
+                    # Extend if the result is a list
+                    results.extend(parsed_results)
+                else:
+                    # Append if it's a single result
+                    results.append(parsed_results)
+
+    return results
+
+
+def parse_addition(data):
+    return {
+        "title": html_to_text(data.get('title', {}).get('content')),
+        "url": data.get('source', {}).get('url'),
+        "content": html_to_text(data.get('summary', {}).get('content')),
+    }
+
+
+def parse_ai_page(data):
+    results = []
+    for item in data.get('list', []):
+        content = (
+            " | ".join(map(str, item.get('content', [])))
+            if isinstance(item.get('content'), list)
+            else str(item.get('content'))
+        )
+        results.append(
+            {
+                "title": html_to_text(item.get('title')),
+                "url": item.get('url'),
+                "content": html_to_text(content),
+                "publishedDate": datetime.fromtimestamp(int(item.get('source', {}).get('time'))),
+            }
+        )
+    return results
+
+
+def parse_baike_sc(data):
+    return {
+        "title": html_to_text(data.get('data', {}).get('title')),
+        "url": data.get('data', {}).get('url'),
+        "content": html_to_text(data.get('data', {}).get('abstract')),
+    }
+
+
+def parse_finance_shuidi(data):
+    content = " | ".join(
+        filter(
+            None,
+            [
+                data.get('establish_time'),
+                data.get('company_status'),
+                data.get('controled_type'),
+                data.get('company_type'),
+                data.get('capital'),
+                data.get('address'),
+                data.get('business_scope'),
+            ],
+        )
```

Comment:
```suggestion
      info for info in [
            data.get('establish_time'),
            data.get('company_status'),
            data.get('controled_type'),
            data.get('company_type'),
            data.get('capital'),
            data.get('address'),
            data.get('business_scope'),
        ] if info,
    )
```
I think that's easier to read because not everybody knows what `filter(None, ...)` does.